lib/backend/dynamo/dynamodbbk.go

```diff
@@ -548,6 +548,95 @@ func (b *Backend) Update(ctx context.Context, item backend.Item) (*backend.Lease
 	return backend.NewLease(item), nil
 }
 
+func (b *Backend) Items(ctx context.Context, params backend.IterateParams) iter.Seq2[backend.Item, error] {
+	if params.StartKey.IsZero() {
+		err := trace.BadParameter("missing parameter startKey")
+		return func(yield func(backend.Item, error) bool) { yield(backend.Item{}, err) }
+	}
+	if params.EndKey.IsZero() {
+		err := trace.BadParameter("missing parameter endKey")
+		return func(yield func(backend.Item, error) bool) { yield(backend.Item{}, err) }
+	}
+
+	const (
+		query = "HashKey = :hashKey AND FullPath BETWEEN :fullPath AND :rangeEnd"
+
+		// filter out expired items, otherwise they might show up in the query
+		// http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/howitworks-ttl.html
+		filter = "attribute_not_exists(Expires) OR Expires >= :timestamp"
+	)
+
+	attrV := map[string]interface{}{
+		":fullPath":  prependPrefix(params.StartKey),
+		":hashKey":   hashKey,
+		":timestamp": b.clock.Now().UTC().Unix(),
+		":rangeEnd":  prependPrefix(params.EndKey),
+	}
+
+	av, err := attributevalue.MarshalMap(attrV)
+	if err != nil {
+		return func(yield func(backend.Item, error) bool) { yield(backend.Item{}, err) }
+	}
+
+	input := dynamodb.QueryInput{
+		KeyConditionExpression:    aws.String(query),
+		TableName:                 &b.TableName,
+		ExpressionAttributeValues: av,
+		FilterExpression:          aws.String(filter),
+		ConsistentRead:            aws.Bool(true),
+		ScanIndexForward:          aws.Bool(!params.Descending),
+	}
+	if params.Limit > 0 {
+		input.Limit = aws.Int32(int32(params.Limit))
+	}
+
+	return func(yield func(backend.Item, error) bool) {
+		count := 0
+		defer func() {
+			if count == backend.DefaultRangeLimit {
+				b.logger.WarnContext(ctx, "Range query hit backend limit. (this is a bug!)", "start_key", params.StartKey, "limit", backend.DefaultRangeLimit)
+			}
+		}()
+
+		paginator := dynamodb.NewQueryPaginator(b.svc, &input)
+		for paginator.HasMorePages() {
+			page, err := paginator.NextPage(ctx)
+			if err != nil {
+				yield(backend.Item{}, convertError(err))
+				return
+			}
+
+			for _, itemAttributes := range page.Items {
+				var r record
+				if err := attributevalue.UnmarshalMap(itemAttributes, &r); err != nil {
+					yield(backend.Item{}, convertError(err))
+					return
+				}
+
+				item := backend.Item{
+					Key:      trimPrefix(r.FullPath),
+					Value:    r.Value,
+					Revision: r.Revision,
+				}
+				if r.Expires != nil {
+					item.Expires = time.Unix(*r.Expires, 0).UTC()
+				}
+				if item.Revision == "" {
+					item.Revision = backend.BlankRevision
+				}
+
+				if !yield(item, nil) {
+					return
+				}
+				count++
+				if count == backend.DefaultRangeLimit || (params.Limit != backend.NoLimit && count >= params.Limit) {
```

Comment:
Can't we do `if params.Limit == 0 { params.Limit = backend.DefaultRangeLimit }` at the beginning?